scopes/lanes/ui/hooks/lanes-context/lanes-model.ts

```diff
@@ -0,0 +1,146 @@
+  };
+};
+export type LaneComponentModel = { model: ComponentModel; url: string };
+export type LaneModel = {
+  id: string;
+  scope: string;
+  url: string;
+  name: string;
+  isMerged: boolean | null;
+  components: LaneComponentModel[];
+};
+
+export function mapToLaneModel(laneData: LaneQueryResult, currentScope: ScopeModel, host: LanesHost): LaneModel {
+  const { name, remote, isMerged } = laneData;
+  const laneName = name;
+  const fullName = remote || name;
+  const laneScope = remote?.split('/')[0];
+
+  const components = laneData.components.map((component) => {
+    const componentName = getLaneComponentName(component, host);
+    const componentScope = getLaneComponentScope(component) || laneScope || currentScope.name;
+    const componentModel = ComponentModel.from({
+      id: {
+        name: componentName,
+        version: component.head,
+        scope: componentScope,
+      },
+      displayName: component.id,
+      compositions: [],
+      packageName: '',
+      description: '',
+    });
+    return { model: componentModel, url: getLaneComponentUrl(componentModel.id, fullName) };
+  });
+
+  return {
+    id: fullName,
+    name: laneName,
+    scope: laneScope || currentScope.name,
+    isMerged,
+    url: getLaneUrl(fullName),
+    components,
+  };
+}
+
+export function groupByScope(lanes: LaneModel[]): Map<string, LaneModel[]> {
+  return lanes.reduce((accum, next) => {
+    const { scope } = next;
+    if (!accum.has(scope)) {
+      accum.set(scope, [next]);
+    } else {
+      const existing = accum.get(scope) as LaneModel[];
+      accum.set(scope, [...existing, next]);
+    }
+    return accum;
+  }, new Map<string, LaneModel[]>());
+}
+
+export function groupByComponentHash(
+  lanes: LaneModel[]
+): Map<string, { lane: LaneModel; component: LaneComponentModel }> {
+  return lanes.reduce((accum, lane) => {
+    const { components } = lane;
+    components.forEach((component) => {
+      accum.set(component.model.id.version as string, { lane, component });
+    });
+    return accum;
+  }, new Map<string, { lane: LaneModel; component: LaneComponentModel }>());
+}
+
+export function mapToLanesModel(lanesData: LanesQueryResult, currentScope: ScopeModel, host: LanesHost): LanesModel {
+  const lanesResult = lanesData.lanes?.getLanes || [];
+  const currentLaneName = lanesData.lanes?.getCurrentLaneName;
+  const laneModel = lanesResult.map((result) => mapToLaneModel(result, currentScope, host));
+  const lanesByScope = groupByScope(laneModel);
+  const lanesByComponentHash = groupByComponentHash(laneModel);
+  const currentLane = laneModel.find((lane) => lane.name === currentLaneName);
+  const lanes = {
+    byScope: lanesByScope,
+    list: laneModel,
+    byComponentHash: lanesByComponentHash,
+  };
+  return {
+    lanes,
+    currentLane,
+    host,
+  };
+}
+
+function getLaneComponentName(laneComponent: LaneComponentQueryResult, host?: LanesHost) {
```

Comment:
1) ok you can to document the LaneComponentId.. how it's built, is it compatible with regular component id, and why you've made it like that.
2) I'd make a generator function like `toLaneComponentId(id: ComponentID, options): ComponentIdObj`, instead of the individual functions.
3) you can't rely on having a `.` to figure out if the component has a scope. consult @davidfirst on how to tackle.
4) we should not rely on host everywhere in the code. We need to think of how to make the scope symmetrical to the workspace.

Reply:
Discussed this with David, I am going to update Lanes GQL query to return a BitId obj instead of a string for Lane Components. That is going to fix all these issues.

Reply:
Removed all logic of parsing LaneCompID from the f/e. Now graphql returns a BitId